### PR TITLE
[MLv2] Filter modal — Reset changes when modal is closed

### PR DESCRIPTION
**Pull request by kulyk**
Resets bulk filter modal's `query` state to `initialQuery` if a modal gets closed by hitting Esc or the "Close" button

### To Verfiy

1. Open the Orders table
2. Open the filter modal
3. Add any filter
4. Click Esc
5. Open the modal again
6. Ensure your changes are not applied

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.tsx [hunk 1]
```diff
@@ -91,6 +91,11 @@ export function BulkFilterModal({
     [initialQuery, query],
   );
 
+  const handleClose = useCallback(() => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.tsx:94]: This is a controlled component, so it'd be more appropriate to add an effect that resets unsaved changes when `opened` prop changes.

File: frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.unit.spec.tsx [hunk 2]
```diff
@@ -119,6 +132,21 @@ describe("BulkFilterModal", () => {
     expect(onClose).toHaveBeenCalled();
   });
 
+  it("should reset changes on close", async () => {
+    const { onSubmit } = setup();
+
+    let createdAtShortcut = screen.getByRole("button", { name: "Today" });
+    userEvent.click(createdAtShortcut);
+    expect(createdAtShortcut).toHaveAttribute("aria-selected", "true");
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.unit.spec.tsx:140]: We could also assert that the submit button is **enabled** here

File: frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.unit.spec.tsx [hunk 3]
```diff
@@ -119,6 +132,21 @@ describe("BulkFilterModal", () => {
...
+
+    userEvent.click(screen.getByLabelText("Close"));
+    userEvent.click(screen.getByRole("button", { name: "Show modal" }));
+    createdAtShortcut = await screen.findByRole("button", { name: "Today" });
+    expect(createdAtShortcut).toHaveAttribute("aria-selected", "false");
+
+    expect(onSubmit).not.toHaveBeenCalled();
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.unit.spec.tsx:147]: We should rather assert that the submit button is **disabled** here

File: frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.tsx [hunk 4]
```diff
@@ -91,6 +91,11 @@ export function BulkFilterModal({
...
+    setQuery(initialQuery); // reset unsaved changes
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/filters/BulkFilterModal/BulkFilterModal.tsx:95]: We can avoid the comment:

```tsx
  const resetUnsavedChanges = () => {
    setQuery(initialQuery);
  };
```

**Review by ranquild — APPROVED** [review]

**Pull request merged**